main.go

```diff
@@ -17,219 +17,59 @@ limitations under the License.
-	flag.BoolVar(&enableWebhook, "enable-webhook", true, "Enable the webhook server(s). Defaults to true.")
 	opts := zap.Options{
 		TimeEncoder: zapcore.TimeEncoder(func(t time.Time, enc zapcore.PrimitiveArrayEncoder) {
 			enc.AppendString(t.UTC().Format("2006-01-02T15:04:05.000Z"))
 		}),
 	}
 	opts.BindFlags(flag.CommandLine)
 	flag.Parse()
-
 	logger := zap.New(zap.UseFlagOptions(&opts))
 	ctrl.SetLogger(logger)
 
-	token, err := os.ReadFile(tokenFile)
-	if err != nil {
-		setupLog.Error(err, "unable to read serviceaccount token")
-		os.Exit(1)
-	}
-	ca, err := readCACert(caCertFile)
-	if err != nil {
-		setupLog.Error(err, "unable to read CA cert")
-		os.Exit(1)
-	}
+	setupLog := ctrl.Log.WithName("setup")
+	ctrl.Log.Info("build info", "commit", version.COMMIT)
+	ctrl.Log.Info("using operator namespace", "namespace", opCfg.OperatorNamespace)
+	ctrl.Log.Info("using AgentImage image", "image", opCfg.AgentImage)
 
-	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-		Logger:                 logger,
-		Scheme:                 scheme,
-		MetricsBindAddress:     metricsAddr,
-		Port:                   9443,
-		HealthProbeBindAddress: probeAddr,
-		LeaderElection:         enableLeaderElection,
-		LeaderElectionID:       "94c735b6.olm.openshift.io",
-		Namespace:              operatorNamespace,
-		// Use a non-caching client everywhere. The default split client does not
-		// promise to invalidate the cache during writes (nor does it promise
-		// sequential create/get coherence), and we have code which (probably
-		// incorrectly) assumes a get immediately following a create/update will
-		// return the updated resource. All client consumers will need audited to
-		// ensure they are tolerant of stale data (or we need a cache or client that
-		// makes stronger coherence guarantees).
-		// https://pkg.go.dev/sigs.k8s.io/controller-runtime#hdr-Clients_and_Caches
-		NewClient: func(_ cache.Cache, config *rest.Config, options client.Options, _ ...client.Object) (client.Client, error) {
-			return client.New(config, options)
-		},
-	})
+	kubeConfig := ctrl.GetConfigOrDie()
+	op, err := operator.New(kubeConfig, &opCfg)
 	if err != nil {
-		setupLog.Error(err, "unable to start manager")
+		setupLog.Error(err, "failed to create NodeObservability operator")
 		os.Exit(1)
 	}
-	// KubeAPI client to be used only in order to get the configmap kubelet-serving-ca
-	// from NS openshift-config-managed.
-	// The clusterWideCli is needed because ConfigMaps are namespaced resources, and in the context
-	// of a namespaced operator, the operator only looks for the namespaced resources in its own namespace
-	// see https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.1/pkg/manager#Options => Namespace
-	clusterWideCli, err := client.New(mgr.GetConfig(), client.Options{
```

Comment:
Hi @ayesha54 
Be careful with the clusterWideCli. It seems to still be used - at least in: https://github.com/ayesha54/node-observability-operator/blob/8cfbbbdb62f176b02c4557814818f827457d2537/pkg/operator/controller/nodeobservability/configmap.go#L30